app/services/catalog/copy_portfolio_item.rb

```diff
@@ -25,6 +25,19 @@ def process
     def make_copy
       @portfolio_item.dup.tap do |new_portfolio_item|
         new_portfolio_item.name = new_name(@portfolio_item.name)
+
+        if @portfolio_item.icon_id.present?
+          new_portfolio_item.create_icon!(:restore_to => new_portfolio_item, :image_id => @portfolio_item.icon.image_id)
+        end
+
+        if @portfolio_item.service_plans.any?
```

Comment:
Minor, but I don't feel like the `any?` check is doing much.  The `each` loop below will not run the block if `service_plans` is an empty array, so this check probably is not required.

Reply:
Oh, great point. I'll remove the check and let the no-op happen when they are empty.